<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>The Experiment</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="site_libs/highlight/textmate.css"
      type="text/css" />
<script src="site_libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.9em;
  padding-left: 5px;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
  padding-left: 10px;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"><em>Mnemiopsis leidyi</em></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="experiment.html">Experiment</a>
</li>
<li>
  <a href="transcriptome.html">Transcriptome</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Quality Control
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="analysis8hpaAllR.html">8hpa Control First Pass</a>
    </li>
    <li>
      <a href="analysis8hpa5RR.html">8hpa Control Second Pass</a>
    </li>
    <li>
      <a href="analysis2hpaAllR.html">2hpa Control First Pass</a>
    </li>
    <li>
      <a href="analysis2hpa2RR.html">2hpa Control Second Pass</a>
    </li>
    <li>
      <a href="analysisStepwise.html">Stepwise Test</a>
    </li>
  </ul>
</li>
<li>
  <a href="StoryOfInitialAnalysis.html">Differential Expression</a>
</li>
<li>
  <a href="downloads.html">Downloads</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">The Experiment</h1>

</div>


<div id="overview" class="section level1">
<h1>Overview</h1>
<p>A time series from 1 hour post fertilization to 9 hours post fertilization (hpf) was collected for RNA-seq to understand the shifts occuring in the developing <em>M. leidyi</em> embryonic transcriptome.</p>
</div>
<div id="sample-collection" class="section level1">
<h1>Sample Collection</h1>
<p>Animals were collected in the wild at various locations around Woods Hole, MA either by net or by a plastic beaker attached to a pole for gently scooping the animals from the water as pictured:</p>
<p><img src="methods/Cteno_catch.jpg" style="width:300px;height:400px;"></p>
<p>Three biological replicates were collected for timepoints 1hpf, 2hpf, 3hpf, 4hpf, 5hpf, 6hpf, 7hpf, 8hpf, 9hpf.</p>
</div>
<div id="rna-seq" class="section level1">
<h1>RNA-seq</h1>
<p>Total RNA was harvested according to</p>
<p><a href="methods/ASA_Trizol-RNA-extraction-Mnemiopsis-V2.pdf">this protocol</a></p>
<p>mRNA was extracted from the total RNA in to parts: <a href="methods/Dynabead-protocol-Mnemiopsis-partI.pdf">part 1</a> and <a href="methods/Dynabead-protocol-Mnemiopsis-partII.pdf">part2</a></p>
<p>Library prep method and sequencing info.</p>
<p>Transcriptome assembly was performed with Trinity using all of the reads for all samples and replicates.</p>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<p>Bowtie2 with the –norc parameter was used to align reads from each developmental timepoint to the reference transcriptome <span class="citation">(Langmead and Salzberg 2012)</span>. Samtools quantified the alignments for each reference sequence <span class="citation">(Li et al. 2009)</span>.</p>
<div id="fastq-processing-commands" class="section level2">
<h2>FASTQ processing commands</h2>
<pre class="sh"><code>
index=&#39;/path/to/transcriptome_index&#39;
fastq=&#39;/path/to/fastq_file&#39;

bowtie2 -p 8 --norc -x ${index} ${fastq}  2&gt;&gt;alignment_report.txt | \

samtools view -@ 1 -b - | samtools sort -T ${fastq}_temp -o ${fastq}.bam -O bam -@ 1 - ;

samtools index ${fastq}.bam;

samtools idxstats ${fastq}.bam | sort -o ${fasq}.idxstats /dev/stdin
</code></pre>
</div>
<div id="r-analysis" class="section level2">
<h2>R Analysis</h2>
<p>Counts for trinity isoforms were summed to create a count ‘per gene’, which served as input to the R package DESeq2 for differential expression analysis and normalization of the read counts <span class="citation">(Love, Huber, and Anders 2014)</span>.</p>
<p>For the differential expression analysis, all timepoints were contrasted with the eight hour time point (omitting the following samples for failing quality control tests: 7hpf_2012_N1’, ‘7hpf_2013_17’, ‘2hpf_2012_N6’, ‘1hpf_2012_N7’, ‘6hpf_2012_N2’), and the two hour time point (omitting the following samples for failing quality control tests: 7hpf_2012_N1’, ‘2hpf_2012_N6’).</p>
<p>All heatmaps were created using the R package pheatmap <span class="citation">(Kolde 2015)</span>.</p>
<p>Gene Ontology <span class="citation">(Consortium and others 2015; Ashburner et al. 2000)</span> terms were assigned to each <em>M. leidyi</em> gene based on homologous PFAM domains and significant Swissprot blastp hits <span class="citation">(Madden 2013)</span>. These terms were used to impute functional categories of differentially expressed genes.</p>
</div>
<div id="geo-submission" class="section level2">
<h2>GEO submission</h2>
<p>Gene expression data has been deposited in the Gene Expression Omnibus (GEO) under the accession [].</p>
</div>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-ashburner2000gene">
<p>Ashburner, Michael, Catherine A Ball, Judith A Blake, David Botstein, Heather Butler, J Michael Cherry, Allan P Davis, et al. 2000. “Gene Ontology: Tool for the Unification of Biology.” <em>Nature Genetics</em> 25 (1). Nature Publishing Group: 25–29.</p>
</div>
<div id="ref-gene2015gene">
<p>Consortium, Gene Ontology, and others. 2015. “Gene Ontology Consortium: Going Forward.” <em>Nucleic Acids Research</em> 43 (D1). Oxford Univ Press: D1049–D1056.</p>
</div>
<div id="ref-kolde2015pheatmap">
<p>Kolde, R. 2015. “Pheatmap: Pretty Heatmaps. R Package Version 1.0. 2.”</p>
</div>
<div id="ref-langmead2012fast">
<p>Langmead, Ben, and Steven L Salzberg. 2012. “Fast Gapped-Read Alignment with Bowtie 2.” <em>Nature Methods</em> 9 (4). Nature Publishing Group: 357–59.</p>
</div>
<div id="ref-li2009sequence">
<p>Li, Heng, Bob Handsaker, Alec Wysoker, Tim Fennell, Jue Ruan, Nils Homer, Gabor Marth, Goncalo Abecasis, Richard Durbin, and others. 2009. “The Sequence Alignment/Map Format and Samtools.” <em>Bioinformatics</em> 25 (16). Oxford Univ Press: 2078–9.</p>
</div>
<div id="ref-love2014moderated">
<p>Love, Michael I, Wolfgang Huber, and Simon Anders. 2014. “Moderated Estimation of Fold Change and Dispersion for Rna-Seq Data with Deseq2.” <em>Genome Biology</em> 15 (12). BioMed Central: 1.</p>
</div>
<div id="ref-madden2013blast">
<p>Madden, Thomas. 2013. “The Blast Sequence Analysis Tool.” National Center for Biotechnology Information (US).</p>
</div>
</div>
</div>


<p>
<a href="http://stowers.org"><img src="images/Stowers-logo_001.jpg" style="width:200px;height:75px;"></a> 
<a href="http://planaria.stowers.org/"><img src="http://smedgd.stowers.org/wp-content/uploads/2015/11/sllogo_whitebg_small.png"></a> 
<a href="http://www.hhmi.org/"><img src="images/HHMI-logo-color_small.jpg"></a>
<a href="http://www.mbl.edu/"><img src="images/marine-biological-laboratory_small.jpg"></a>
</p>



<p><a href="http://www.stowers.org/privacy-policy">Privacy Policy</a></p>
<p><a href="http://www.stowers.org/terms">Terms of Use</a></p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
