<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Eric Ross" />


<title>Transcriptome</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="site_libs/highlight/textmate.css"
      type="text/css" />
<script src="site_libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
#rmd-source-code {
  display: none;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: -22px; }
.code-folding-btn.collapsed { margin-bottom: 4px; }
</style>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<script src="site_libs/navigation-1.1/FileSaver.min.js"></script>
<script src="site_libs/navigation-1.1/sourceembed.js"></script>
<script>
$(document).ready(function () {
  window.initializeSourceEmbed("transcriptome.Rmd");
});
</script>




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.9em;
  padding-left: 5px;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
  padding-left: 10px;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"><em>Mnemiopsis leidyi</em></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="experiment.html">Experiment</a>
</li>
<li>
  <a href="transcriptome.html">Transcriptome</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Quality Control
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="analysis8hpaAllR.html">8hpa Control First Pass</a>
    </li>
    <li>
      <a href="analysis8hpa5RR.html">8hpa Control Second Pass</a>
    </li>
    <li>
      <a href="analysis2hpaAllR.html">2hpa Control First Pass</a>
    </li>
    <li>
      <a href="analysis2hpa2RR.html">2hpa Control Second Pass</a>
    </li>
    <li>
      <a href="analysisStepwise.html">Stepwise Test</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Differential Expression
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="StoryOfInitialAnalysis.html">Summary Analysis</a>
    </li>
    <li>
      <a href="InitialAnalysis.html">Full Analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="downloads.html">Downloads</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->


<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-download-source" href="#">Download Rmd</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Transcriptome</h1>
<h4 class="author"><em>Eric Ross</em></h4>

</div>


<div id="overview" class="section level1">
<h1>Overview</h1>
<p>Assembly and analysis of <em>Mnemiopsis leidyi</em> transcriptome.</p>
</div>
<div id="data-sources" class="section level1">
<h1>Data Sources</h1>
<p>Sequencing Directory: /n/projects/ejr/external_sequencing/Mnemiopsis_leidyi/NCBI</p>
<table>
<thead>
<tr class="header">
<th>Time point</th>
<th>Year Collected</th>
<th>Replicate Name</th>
<th>left/right</th>
<th>Filename</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1hpf</td>
<td>2012</td>
<td>N7</td>
<td>1</td>
<td>1hpf_2012_N7_1.fastq.gz</td>
</tr>
<tr class="even">
<td>1hpf</td>
<td>2012</td>
<td>N7</td>
<td>2</td>
<td>1hpf_2012_N7_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>1hpf</td>
<td>2013</td>
<td>21</td>
<td>1</td>
<td>1hpf_2013_21_1.fastq.gz</td>
</tr>
<tr class="even">
<td>1hpf</td>
<td>2013</td>
<td>21</td>
<td>2</td>
<td>1hpf_2013_21_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>1hpf</td>
<td>2013</td>
<td>41</td>
<td>1</td>
<td>1hpf_2013_41_1.fastq.gz</td>
</tr>
<tr class="even">
<td>1hpf</td>
<td>2013</td>
<td>41</td>
<td>2</td>
<td>1hpf_2013_41_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>2hpf</td>
<td>2012</td>
<td>N6</td>
<td>1</td>
<td>2hpf_2012_N6_1.fastq.gz</td>
</tr>
<tr class="even">
<td>2hpf</td>
<td>2012</td>
<td>N6</td>
<td>2</td>
<td>2hpf_2012_N6_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>2hpf</td>
<td>2013</td>
<td>22</td>
<td>1</td>
<td>2hpf_2013_22_1.fastq.gz</td>
</tr>
<tr class="even">
<td>2hpf</td>
<td>2013</td>
<td>22</td>
<td>2</td>
<td>2hpf_2013_22_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>2hpf</td>
<td>2013</td>
<td>32</td>
<td>1</td>
<td>2hpf_2013_32_1.fastq.gz</td>
</tr>
<tr class="even">
<td>2hpf</td>
<td>2013</td>
<td>32</td>
<td>2</td>
<td>2hpf_2013_32_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>2hpf</td>
<td>2013</td>
<td>42</td>
<td>1</td>
<td>2hpf_2013_42_1.fastq.gz</td>
</tr>
<tr class="even">
<td>2hpf</td>
<td>2013</td>
<td>42</td>
<td>2</td>
<td>2hpf_2013_42_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>3hpf</td>
<td>2012</td>
<td>N5</td>
<td>1</td>
<td>3hpf_2012_N5_1.fastq.gz</td>
</tr>
<tr class="even">
<td>3hpf</td>
<td>2012</td>
<td>N5</td>
<td>2</td>
<td>3hpf_2012_N5_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>3hpf</td>
<td>2013</td>
<td>23</td>
<td>1</td>
<td>3hpf_2013_23_1.fastq.gz</td>
</tr>
<tr class="even">
<td>3hpf</td>
<td>2013</td>
<td>23</td>
<td>2</td>
<td>3hpf_2013_23_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>3hpf</td>
<td>2013</td>
<td>33</td>
<td>1</td>
<td>3hpf_2013_33_1.fastq.gz</td>
</tr>
<tr class="even">
<td>3hpf</td>
<td>2013</td>
<td>33</td>
<td>2</td>
<td>3hpf_2013_33_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>4hpf</td>
<td>2012</td>
<td>N4</td>
<td>1</td>
<td>4hpf_2012_N4_1.fastq.gz</td>
</tr>
<tr class="even">
<td>4hpf</td>
<td>2012</td>
<td>N4</td>
<td>2</td>
<td>4hpf_2012_N4_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>4hpf</td>
<td>2013</td>
<td>24</td>
<td>1</td>
<td>4hpf_2013_24_1.fastq.gz</td>
</tr>
<tr class="even">
<td>4hpf</td>
<td>2013</td>
<td>24</td>
<td>2</td>
<td>4hpf_2013_24_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>4hpf</td>
<td>2013</td>
<td>34</td>
<td>1</td>
<td>4hpf_2013_34_1.fastq.gz</td>
</tr>
<tr class="even">
<td>4hpf</td>
<td>2013</td>
<td>34</td>
<td>2</td>
<td>4hpf_2013_34_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>5hpf</td>
<td>2012</td>
<td>14</td>
<td>1</td>
<td>5hpf_2012_14_1.fastq.gz</td>
</tr>
<tr class="even">
<td>5hpf</td>
<td>2012</td>
<td>14</td>
<td>2</td>
<td>5hpf_2012_14_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>5hpf</td>
<td>2012</td>
<td>N3</td>
<td>1</td>
<td>5hpf_2012_N3_1.fastq.gz</td>
</tr>
<tr class="even">
<td>5hpf</td>
<td>2012</td>
<td>N3</td>
<td>2</td>
<td>5hpf_2012_N3_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>5hpf</td>
<td>2013</td>
<td>25</td>
<td>1</td>
<td>5hpf_2013_25_1.fastq.gz</td>
</tr>
<tr class="even">
<td>5hpf</td>
<td>2013</td>
<td>25</td>
<td>2</td>
<td>5hpf_2013_25_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>6hpf</td>
<td>2012</td>
<td>23</td>
<td>1</td>
<td>6hpf_2012_23_1.fastq.gz</td>
</tr>
<tr class="even">
<td>6hpf</td>
<td>2012</td>
<td>23</td>
<td>2</td>
<td>6hpf_2012_23_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>6hpf</td>
<td>2012</td>
<td>N2</td>
<td>1</td>
<td>6hpf_2012_N2_1.fastq.gz</td>
</tr>
<tr class="even">
<td>6hpf</td>
<td>2012</td>
<td>N2</td>
<td>2</td>
<td>6hpf_2012_N2_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>6hpf</td>
<td>2013</td>
<td>26</td>
<td>1</td>
<td>6hpf_2013_26_1.fastq.gz</td>
</tr>
<tr class="even">
<td>6hpf</td>
<td>2013</td>
<td>26</td>
<td>2</td>
<td>6hpf_2013_26_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>6hpf</td>
<td>2013</td>
<td>46</td>
<td>1</td>
<td>6hpf_2013_46_1.fastq.gz</td>
</tr>
<tr class="even">
<td>6hpf</td>
<td>2013</td>
<td>46</td>
<td>2</td>
<td>6hpf_2013_46_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>7hpf</td>
<td>2012</td>
<td>N1</td>
<td>1</td>
<td>7hpf_2012_N1_1.fastq.gz</td>
</tr>
<tr class="even">
<td>7hpf</td>
<td>2012</td>
<td>N1</td>
<td>2</td>
<td>7hpf_2012_N1_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>7hpf</td>
<td>2013</td>
<td>17</td>
<td>1</td>
<td>7hpf_2013_17_1.fastq.gz</td>
</tr>
<tr class="even">
<td>7hpf</td>
<td>2013</td>
<td>17</td>
<td>2</td>
<td>7hpf_2013_17_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>7hpf</td>
<td>2013</td>
<td>27</td>
<td>1</td>
<td>7hpf_2013_27_1.fastq.gz</td>
</tr>
<tr class="even">
<td>7hpf</td>
<td>2013</td>
<td>27</td>
<td>2</td>
<td>7hpf_2013_27_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>7hpf</td>
<td>2013</td>
<td>47</td>
<td>1</td>
<td>7hpf_2013_47_1.fastq.gz</td>
</tr>
<tr class="even">
<td>7hpf</td>
<td>2013</td>
<td>47</td>
<td>2</td>
<td>7hpf_2013_47_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>8hpf</td>
<td>2012</td>
<td>18</td>
<td>1</td>
<td>8hpf_2012_18_1.fastq.gz</td>
</tr>
<tr class="even">
<td>8hpf</td>
<td>2012</td>
<td>18</td>
<td>2</td>
<td>8hpf_2012_18_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>8hpf</td>
<td>2012</td>
<td>N8B</td>
<td>1</td>
<td>8hpf_2012_N8B_1.fastq.gz</td>
</tr>
<tr class="even">
<td>8hpf</td>
<td>2012</td>
<td>N8B</td>
<td>2</td>
<td>8hpf_2012_N8B_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>8hpf</td>
<td>2013</td>
<td>18</td>
<td>1</td>
<td>8hpf_2013_18_1.fastq.gz</td>
</tr>
<tr class="even">
<td>8hpf</td>
<td>2013</td>
<td>18</td>
<td>2</td>
<td>8hpf_2013_18_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>8hpf</td>
<td>2013</td>
<td>28</td>
<td>1</td>
<td>8hpf_2013_28_1.fastq.gz</td>
</tr>
<tr class="even">
<td>8hpf</td>
<td>2013</td>
<td>28</td>
<td>2</td>
<td>8hpf_2013_28_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>8hpf</td>
<td>2013</td>
<td>48</td>
<td>1</td>
<td>8hpf_2013_48_1.fastq.gz</td>
</tr>
<tr class="even">
<td>8hpf</td>
<td>2013</td>
<td>48</td>
<td>2</td>
<td>8hpf_2013_48_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>9hpf</td>
<td>2012</td>
<td>37</td>
<td>1</td>
<td>9hpf_2012_37_1.fastq.gz</td>
</tr>
<tr class="even">
<td>9hpf</td>
<td>2012</td>
<td>37</td>
<td>2</td>
<td>9hpf_2012_37_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>9hpf</td>
<td>2013</td>
<td>49</td>
<td>1</td>
<td>9hpf_2013_49_1.fastq.gz</td>
</tr>
<tr class="even">
<td>9hpf</td>
<td>2013</td>
<td>49</td>
<td>2</td>
<td>9hpf_2013_49_2.fastq.gz</td>
</tr>
</tbody>
</table>
</div>
<div id="assembly" class="section level1">
<h1>Assembly</h1>
<p>Transcriptome assembly was performed with Trinity (trinityrnaseq_r20140717). Left and right reads were concatenated and then passed to Trinity.</p>
<pre><code>Trinity \  
--normalize_reads \  
--max_memory 500G \  
--CPU 16 \  
--SS_lib_type FR \  
--left \  
lefts.fastq \  
--right \  
rights.fastq  
</code></pre>
<div id="initial-assembly-statistics" class="section level2">
<h2>Initial Assembly Statistics</h2>
<table>
<tbody>
<tr class="odd">
<td>Number of Sequences:</td>
<td align="right">233,327</td>
</tr>
<tr class="even">
<td>Total Length:</td>
<td align="right">160,677,750</td>
</tr>
<tr class="odd">
<td>Average Length:</td>
<td align="right">688</td>
</tr>
<tr class="even">
<td>Longest Sequence:</td>
<td align="right">29,348</td>
</tr>
<tr class="odd">
<td>Shortest Sequence:</td>
<td align="right">201</td>
</tr>
<tr class="even">
<td>%GC:</td>
<td align="right">40%</td>
</tr>
<tr class="odd">
<td>N50:</td>
<td align="right">1,152</td>
</tr>
<tr class="even">
<td>N90:</td>
<td align="right">266</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="sequence-cleaning-round-1" class="section level1">
<h1>Sequence Cleaning (Round 1)</h1>
<p>Assembled Transcriptome was run through seqclean to clean up adapter and contaminate from other species. Seqclean also removes sequence below the specified length (300bp). Trinity defaults to a 200bp minimum length which caused 90,500 short sequences to be dropped.</p>
<pre><code>seqclean \  
Mnemiopsis_leidyi_trinity_20140802_unfiltered.nt \  
-v /n/projects/ejr/blastdb/contaminate/UniVec \  
-l 300 \  
-c 16 \  
-s /n/projects/ejr/blastdb/contaminate/Hs_GRCh38.nt,/n/projects/ejr/blastdb/contaminate/ecoli.nt  
</code></pre>
<pre><code>Output:
**************************************************
Sequences analyzed:    233327
-----------------------------------
                   valid:    141346  (24371 trimmed)
                 trashed:     91981
**************************************************
----= Trashing summary =------
               by 'short':    90500
        by 'Hs_GRCh38.nt':       11
              by 'shortq':      140
            by 'ecoli.nt':       17
              by 'UniVec':     1313
------------------------------
</code></pre>
<div id="cleaned-assembly-statistics" class="section level2">
<h2>Cleaned Assembly Statistics</h2>
<table>
<tbody>
<tr class="odd">
<td>Number of Sequences:</td>
<td align="right">141,346</td>
</tr>
<tr class="even">
<td>Total Length:</td>
<td align="right">137,906,212</td>
</tr>
<tr class="odd">
<td>Average Length:</td>
<td align="right">975</td>
</tr>
<tr class="even">
<td>Longest Sequence:</td>
<td align="right">29,348</td>
</tr>
<tr class="odd">
<td>Shortest Sequence:</td>
<td align="right">300</td>
</tr>
<tr class="even">
<td>%GC:</td>
<td align="right">41%</td>
</tr>
<tr class="odd">
<td>N50:</td>
<td align="right">1,486</td>
</tr>
<tr class="even">
<td>N90:</td>
<td align="right">405</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="genome-filtering" class="section level1">
<h1>Genome Filtering</h1>
<p>In order to produce a more accurate transcriptome set we filtered out sequences not present in the available <em>M. leidyi</em> genome. We used a very permissive requirement that any transcript alignign with at least 90% identity over any length of alignment ought to be retained.</p>
<div id="genome-filtered-transcript-statistics" class="section level2">
<h2>Genome Filtered Transcript Statistics</h2>
<table>
<tbody>
<tr class="odd">
<td>Number of Sequences:</td>
<td align="right">44,438</td>
</tr>
<tr class="even">
<td>Total Length:</td>
<td align="right">62,917,788</td>
</tr>
<tr class="odd">
<td>Average Length:</td>
<td align="right">1,415</td>
</tr>
<tr class="even">
<td>Longest Sequence:</td>
<td align="right">29,348</td>
</tr>
<tr class="odd">
<td>Shortest Sequence:</td>
<td align="right">300</td>
</tr>
<tr class="even">
<td>%GC:</td>
<td align="right">41%</td>
</tr>
<tr class="odd">
<td>%N:</td>
<td align="right">0%</td>
</tr>
<tr class="even">
<td>N50:</td>
<td align="right">2,168</td>
</tr>
<tr class="odd">
<td>N90:</td>
<td align="right">601</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="sequence-cleaning-additional-rounds" class="section level1">
<h1>Sequence Cleaning (Additional Rounds)</h1>
<p>NCBI detected additional contaminate in our sequences, primarily vector. Seqclean did not remove concatenated adapter. We ran additional rounds of Seqclean until we could no longer detect adapter.</p>
<div id="cleaned-sequence-statistics" class="section level2">
<h2>Cleaned Sequence Statistics</h2>
<table>
<tbody>
<tr class="odd">
<td>Number of Sequences:</td>
<td align="right">44,388</td>
</tr>
<tr class="even">
<td>Total Length:</td>
<td align="right">62,859,653</td>
</tr>
<tr class="odd">
<td>Average Length:</td>
<td align="right">1,416</td>
</tr>
<tr class="even">
<td>Longest Sequence:</td>
<td align="right">29,348</td>
</tr>
<tr class="odd">
<td>Shortest Sequence:</td>
<td align="right">300</td>
</tr>
<tr class="even">
<td>%GC:</td>
<td align="right">41%</td>
</tr>
<tr class="odd">
<td>%N:</td>
<td align="right">0%</td>
</tr>
<tr class="even">
<td>N50:</td>
<td align="right">2,167</td>
</tr>
<tr class="odd">
<td>N90:</td>
<td align="right">601</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="transcript-collapse" class="section level1">
<h1>Transcript collapse</h1>
<p>For RNAseq we wished to create a non-redundant set of transcripts by collapsing isoforms using CD-HIT.</p>
<pre><code>cd-hit-est \  
-i in_genome_clean_r6.nt \  
-o cdhit.nt \  
-r 0 \  
-c .95 \  
-G 0 \  
-aS .5 \  
-M 0 \  
-T 0  
</code></pre>
<div id="non-redundant-sequence-statistics" class="section level2">
<h2>Non-redundant Sequence Statistics</h2>
<table>
<tbody>
<tr class="odd">
<td>Number of Sequences:</td>
<td align="right">31,067</td>
</tr>
<tr class="even">
<td>Total Length:</td>
<td align="right">47,344,147</td>
</tr>
<tr class="odd">
<td>Average Length:</td>
<td align="right">1,523</td>
</tr>
<tr class="even">
<td>Longest Sequence:</td>
<td align="right">29,348</td>
</tr>
<tr class="odd">
<td>Shortest Sequence:</td>
<td align="right">300</td>
</tr>
<tr class="even">
<td>%GC:</td>
<td align="right">41%</td>
</tr>
<tr class="odd">
<td>%N:</td>
<td align="right">0%</td>
</tr>
<tr class="even">
<td>N50:</td>
<td align="right">2,200</td>
</tr>
<tr class="odd">
<td>N90:</td>
<td align="right">692</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="sequence-renaming" class="section level1">
<h1>Sequence Renaming</h1>
<p>Sequence were renamed with a “Mle” prefix and zero padded numbers. Names are in the format of Mle_000001.</p>
</div>
<div id="annotation" class="section level1">
<h1>Annotation</h1>
<div id="swissprot-uniprot" class="section level2">
<h2>Swissprot / uniprot</h2>
<p>Best BLASTX hits to swissprot with a maximum evalue of .001 were used to annotate genes.</p>
<pre><code>blastx \
-query Mleidyi_20160525.nt \  
-db swissprot \  
-max_target_seqs 1 \  
-max_hsps 1 \  
-evalue .001 \  
-outfmt 6 \  
-out ml_v_sp.blastx \  
-num_threads 32
</code></pre>
</div>
<div id="pfam" class="section level2">
<h2>Pfam</h2>
Sequences were translated with Transdecoder and then Domains were assigned to transcripts using hmmscan against Pfam-A.<br />

<pre><code>Transdecoder.LongOrfs \  
-S \
-m 60 \
-t Mleidyi_20160525.aa.fa
</code></pre>
<p>All ORFs of 60aa or greater were retained.</p>
<pre><code>hmmscan \  
-o Mleidyi_20160525.aa.fa.pfam.out \  
--domtblout Mleidyi_20160525.aa.fa.pfam.table \  
-E .01 \  
--domE .01 \  
--cpu 16 \  
../db/Pfam-A.hmm \  
</code></pre>
<pre><code>
awk '{print $4 "\t" $2}' Mleidyi_20160525.aa.fa.pfam.table | perl -p -e 's/(Mle_\d+)\|.*(\t.+)/$1$2/' | sort -u > mleidyi2pfam.txt
</code></pre>
</div>
<div id="software" class="section level2">
<h2>Software</h2>
<table>
<thead>
<tr class="header">
<th>Program</th>
<th>Version</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Trinity</td>
<td>trinityrnaseq_r20140717</td>
</tr>
<tr class="even">
<td>Seqclean</td>
<td>Downloaded Feb 22, 2011</td>
</tr>
<tr class="odd">
<td>BLAST+</td>
<td>2.3.0</td>
</tr>
<tr class="even">
<td>Transdecoder</td>
<td>2.0.1</td>
</tr>
<tr class="odd">
<td>cd-hit</td>
<td>CD-HIT version 4.6</td>
</tr>
<tr class="even">
<td>HMMeR</td>
<td>3.1b2</td>
</tr>
</tbody>
</table>
</div>
<div id="accession-numbers" class="section level2">
<h2>Accession Numbers</h2>
<p>BioProject: PRJNA344880<br />
TSA: GFAT00000000</p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>CDHIT:<br />
“Clustering of highly homologous sequences to reduce thesize of large protein database”, Weizhong Li, Lukasz Jaroszewski &amp; Adam Godzik. Bioinformatics, (2001) 17:282-283<br />
“Cd-hit: a fast program for clustering and comparing large sets of protein or nucleotide sequences”, Weizhong Li &amp; Adam Godzik. Bioinformatics, (2006) 22:1658-1659</p>
<p>HMMER:<br />
Accelerated profile HMM searches. S. R. Eddy. PLoS Comp. Biol., 7:e1002195, 2011.</p>
<p>Trinity and Transdecoder:<br />
Grabherr MG, Haas BJ, Yassour M, Levin JZ, Thompson DA, Amit I, Adiconis X, Fan L, Raychowdhury R, Zeng Q, Chen Z, Mauceli E, Hacohen N, Gnirke A, Rhind N, di Palma F, Birren BW, Nusbaum C, Lindblad-Toh K, Friedman N, Regev A. Full-length transcriptome assembly from RNA-seq data without a reference genome. Nat Biotechnol. 2011 May 15;29(7):644-52. doi: 10.1038/nbt.1883. PubMed PMID: 21572440.</p>
<p>BLAST+:<br />
Camacho C., Coulouris G., Avagyan V., Ma N., Papadopoulos J., Bealer K., &amp; Madden T.L. (2008) “BLAST+: architecture and applications.” BMC Bioinformatics 10:421.</p>
<p>Seqclean:<br />
<a href="https://sourceforge.net/projects/seqclean/" class="uri">https://sourceforge.net/projects/seqclean/</a></p>
</div>
</div>
<div id="r-session-information" class="section level1">
<h1><a href="http://bioinfo/~klg/projects/Mnemiopsis/website/sessionInfo/StorySessionInfo.txt">R-session information</a></h1>
</div>

<div id="rmd-source-code">---
title: "Transcriptome" 
author: "Eric Ross"
output:
  html_document:
      code_download: true
---

# Overview

Assembly and analysis of _Mnemiopsis leidyi_ transcriptome.

```{r setup, include = FALSE}
# better rendering error messages
library(knitr)
opts_chunk$set(message=FALSE, results = 'hide', fig.keep="all", warning=FALSE, error=FALSE, fig.path="./analysis_figures/")
library(tidyverse)
library(ggthemes)
library(pander)
library(gridExtra)
library(RColorBrewer)
library(wesanderson)
library(gplots)
library(rbokeh)
  
setwd("/home/ejr/analysis/")
```

# Data Sources

Sequencing Directory: /n/projects/ejr/external_sequencing/Mnemiopsis_leidyi/NCBI

| Time point | Year Collected | Replicate Name | left/right | Filename |
| ---- | ---- | -- | - | ----------------------- |
| 1hpf | 2012 | N7 | 1 | 1hpf_2012_N7_1.fastq.gz |
| 1hpf | 2012 | N7 | 2 | 1hpf_2012_N7_2.fastq.gz |
| 1hpf | 2013 | 21 | 1 | 1hpf_2013_21_1.fastq.gz |
| 1hpf | 2013 | 21 | 2 | 1hpf_2013_21_2.fastq.gz |
| 1hpf | 2013 | 41 | 1 | 1hpf_2013_41_1.fastq.gz |
| 1hpf | 2013 | 41 | 2 | 1hpf_2013_41_2.fastq.gz |
| 2hpf | 2012 | N6 | 1 | 2hpf_2012_N6_1.fastq.gz |
| 2hpf | 2012 | N6 | 2 | 2hpf_2012_N6_2.fastq.gz |
| 2hpf | 2013 | 22 | 1 | 2hpf_2013_22_1.fastq.gz |
| 2hpf | 2013 | 22 | 2 | 2hpf_2013_22_2.fastq.gz |
| 2hpf | 2013 | 32 | 1 | 2hpf_2013_32_1.fastq.gz |
| 2hpf | 2013 | 32 | 2 | 2hpf_2013_32_2.fastq.gz |
| 2hpf | 2013 | 42 | 1 | 2hpf_2013_42_1.fastq.gz |
| 2hpf | 2013 | 42 | 2 | 2hpf_2013_42_2.fastq.gz |
| 3hpf | 2012 | N5 | 1 | 3hpf_2012_N5_1.fastq.gz |
| 3hpf | 2012 | N5 | 2 | 3hpf_2012_N5_2.fastq.gz |
| 3hpf | 2013 | 23 | 1 | 3hpf_2013_23_1.fastq.gz |
| 3hpf | 2013 | 23 | 2 | 3hpf_2013_23_2.fastq.gz |
| 3hpf | 2013 | 33 | 1 | 3hpf_2013_33_1.fastq.gz |
| 3hpf | 2013 | 33 | 2 | 3hpf_2013_33_2.fastq.gz |
| 4hpf | 2012 | N4 | 1 | 4hpf_2012_N4_1.fastq.gz |
| 4hpf | 2012 | N4 | 2 | 4hpf_2012_N4_2.fastq.gz |
| 4hpf | 2013 | 24 | 1 | 4hpf_2013_24_1.fastq.gz |
| 4hpf | 2013 | 24 | 2 | 4hpf_2013_24_2.fastq.gz |
| 4hpf | 2013 | 34 | 1 | 4hpf_2013_34_1.fastq.gz |
| 4hpf | 2013 | 34 | 2 | 4hpf_2013_34_2.fastq.gz |
| 5hpf | 2012 | 14 | 1 | 5hpf_2012_14_1.fastq.gz |
| 5hpf | 2012 | 14 | 2 | 5hpf_2012_14_2.fastq.gz |
| 5hpf | 2012 | N3 | 1 | 5hpf_2012_N3_1.fastq.gz |
| 5hpf | 2012 | N3 | 2 | 5hpf_2012_N3_2.fastq.gz |
| 5hpf | 2013 | 25 | 1 | 5hpf_2013_25_1.fastq.gz |
| 5hpf | 2013 | 25 | 2 | 5hpf_2013_25_2.fastq.gz |
| 6hpf | 2012 | 23 | 1 | 6hpf_2012_23_1.fastq.gz |
| 6hpf | 2012 | 23 | 2 | 6hpf_2012_23_2.fastq.gz |
| 6hpf | 2012 | N2 | 1 | 6hpf_2012_N2_1.fastq.gz |
| 6hpf | 2012 | N2 | 2 | 6hpf_2012_N2_2.fastq.gz |
| 6hpf | 2013 | 26 | 1 | 6hpf_2013_26_1.fastq.gz |
| 6hpf | 2013 | 26 | 2 | 6hpf_2013_26_2.fastq.gz |
| 6hpf | 2013 | 46 | 1 | 6hpf_2013_46_1.fastq.gz |
| 6hpf | 2013 | 46 | 2 | 6hpf_2013_46_2.fastq.gz |
| 7hpf | 2012 | N1 | 1 | 7hpf_2012_N1_1.fastq.gz |
| 7hpf | 2012 | N1 | 2 | 7hpf_2012_N1_2.fastq.gz |
| 7hpf | 2013 | 17 | 1 | 7hpf_2013_17_1.fastq.gz |
| 7hpf | 2013 | 17 | 2 | 7hpf_2013_17_2.fastq.gz |
| 7hpf | 2013 | 27 | 1 | 7hpf_2013_27_1.fastq.gz |
| 7hpf | 2013 | 27 | 2 | 7hpf_2013_27_2.fastq.gz |
| 7hpf | 2013 | 47 | 1 | 7hpf_2013_47_1.fastq.gz |
| 7hpf | 2013 | 47 | 2 | 7hpf_2013_47_2.fastq.gz |
| 8hpf | 2012 | 18 | 1 | 8hpf_2012_18_1.fastq.gz |
| 8hpf | 2012 | 18 | 2 | 8hpf_2012_18_2.fastq.gz |
| 8hpf | 2012 | N8B | 1 | 8hpf_2012_N8B_1.fastq.gz |
| 8hpf | 2012 | N8B | 2 | 8hpf_2012_N8B_2.fastq.gz |
| 8hpf | 2013 | 18 | 1 | 8hpf_2013_18_1.fastq.gz |
| 8hpf | 2013 | 18 | 2 | 8hpf_2013_18_2.fastq.gz |
| 8hpf | 2013 | 28 | 1 | 8hpf_2013_28_1.fastq.gz |
| 8hpf | 2013 | 28 | 2 | 8hpf_2013_28_2.fastq.gz |
| 8hpf | 2013 | 48 | 1 | 8hpf_2013_48_1.fastq.gz |
| 8hpf | 2013 | 48 | 2 | 8hpf_2013_48_2.fastq.gz |
| 9hpf | 2012 | 37 | 1 | 9hpf_2012_37_1.fastq.gz |
| 9hpf | 2012 | 37 | 2 | 9hpf_2012_37_2.fastq.gz |
| 9hpf | 2013 | 49 | 1 | 9hpf_2013_49_1.fastq.gz |
| 9hpf | 2013 | 49 | 2 | 9hpf_2013_49_2.fastq.gz |


```{r DataSources, echo=FALSE, warning=FALSE, message=FALSE, fig.width=10}
# figure on depth here?
```

# Assembly

Transcriptome assembly was performed with Trinity (trinityrnaseq_r20140717).  Left and right reads were concatenated and then passed to Trinity.

<pre><code>Trinity \  
--normalize_reads \  
--max_memory 500G \  
--CPU 16 \  
--SS_lib_type FR \  
--left \  
lefts.fastq \  
--right \  
rights.fastq  
</code></pre>

## Initial Assembly Statistics

| | |
|---|---:|
|Number of Sequences: | 233,327|
|Total Length: | 160,677,750|
|Average Length: | 688|
|Longest Sequence: | 29,348|
|Shortest Sequence: | 201|
|%GC: | 40%|
|N50: | 1,152|
|N90: | 266|


# Sequence Cleaning (Round 1)

Assembled Transcriptome was run through seqclean to clean up adapter and contaminate from other species.  Seqclean also removes sequence below the specified length (300bp).  Trinity defaults to a 200bp minimum length which caused 90,500 short sequences to be dropped.

<pre><code>seqclean \  
Mnemiopsis_leidyi_trinity_20140802_unfiltered.nt \  
-v /n/projects/ejr/blastdb/contaminate/UniVec \  
-l 300 \  
-c 16 \  
-s /n/projects/ejr/blastdb/contaminate/Hs_GRCh38.nt,/n/projects/ejr/blastdb/contaminate/ecoli.nt  
</code></pre>


<pre><code>Output:
**************************************************
Sequences analyzed:    233327
-----------------------------------
                   valid:    141346  (24371 trimmed)
                 trashed:     91981
**************************************************
----= Trashing summary =------
               by 'short':    90500
        by 'Hs_GRCh38.nt':       11
              by 'shortq':      140
            by 'ecoli.nt':       17
              by 'UniVec':     1313
------------------------------
</code></pre>

## Cleaned Assembly Statistics

| | |
|---|---:|
|Number of Sequences: | 141,346|
|Total Length: | 137,906,212|
|Average Length: | 975|
|Longest Sequence: | 29,348|
|Shortest Sequence: | 300|
|%GC: | 41%|
|N50: | 1,486|
|N90: | 405|


# Genome Filtering

In order to produce a more accurate transcriptome set we filtered out sequences not present in the available _M. leidyi_ genome. We used a very permissive requirement that any transcript alignign with at least 90% identity over any length of alignment ought to be retained.


## Genome Filtered Transcript Statistics

| | |
|---|---:|
|Number of Sequences: | 44,438|
|Total Length: | 62,917,788|
|Average Length: | 1,415|
|Longest Sequence: | 29,348|
|Shortest Sequence: | 300|
|%GC: | 41%|
|%N: | 0%|
|N50: | 2,168|
|N90: | 601|

# Sequence Cleaning (Additional Rounds)

NCBI detected additional contaminate in our sequences, primarily vector.  Seqclean did not remove concatenated adapter.  We ran additional rounds of Seqclean until we could no longer detect adapter.

## Cleaned Sequence Statistics

| | |
|---|---:|
|Number of Sequences: | 44,388|
|Total Length: | 62,859,653|
|Average Length: | 1,416|
|Longest Sequence: | 29,348|
|Shortest Sequence: | 300|
|%GC: | 41%|
|%N: | 0%|
|N50: | 2,167|
|N90: | 601|

# Transcript collapse

For RNAseq we wished to create a non-redundant set of transcripts by collapsing isoforms using CD-HIT.

<pre><code>cd-hit-est \  
-i in_genome_clean_r6.nt \  
-o cdhit.nt \  
-r 0 \  
-c .95 \  
-G 0 \  
-aS .5 \  
-M 0 \  
-T 0  
</code></pre>

## Non-redundant Sequence Statistics

| | |
|---|---:|
|Number of Sequences: | 31,067|
|Total Length: | 47,344,147|
|Average Length: | 1,523|
|Longest Sequence: | 29,348|
|Shortest Sequence: | 300|
|%GC: | 41%|
|%N: | 0%|
|N50: | 2,200|
|N90: | 692|


# Sequence Renaming

Sequence were renamed with a "Mle" prefix and zero padded numbers.  Names are in the format of Mle_000001.  

# Annotation

## Swissprot / uniprot

Best BLASTX hits to swissprot with a maximum evalue of .001 were used to annotate genes.  

<pre><code>blastx \
-query Mleidyi_20160525.nt \  
-db swissprot \  
-max_target_seqs 1 \  
-max_hsps 1 \  
-evalue .001 \  
-outfmt 6 \  
-out ml_v_sp.blastx \  
-num_threads 32
</code></pre>

## Pfam
 
Sequences were translated with Transdecoder and then Domains were assigned to transcripts using hmmscan against Pfam-A.  
<pre><code>Transdecoder.LongOrfs \  
-S \
-m 60 \
-t Mleidyi_20160525.aa.fa
</code></pre>

All ORFs of 60aa or greater were retained.

<pre><code>hmmscan \  
-o Mleidyi_20160525.aa.fa.pfam.out \  
--domtblout Mleidyi_20160525.aa.fa.pfam.table \  
-E .01 \  
--domE .01 \  
--cpu 16 \  
../db/Pfam-A.hmm \  
</code></pre>


<pre><code>
awk '{print $4 "\t" $2}' Mleidyi_20160525.aa.fa.pfam.table | perl -p -e 's/(Mle_\d+)\|.*(\t.+)/$1$2/' | sort -u > mleidyi2pfam.txt
</code></pre>

## Software


| Program      | Version                   |
| ------------ | --------------------------|
| Trinity      | trinityrnaseq_r20140717   |
| Seqclean     | Downloaded Feb 22, 2011   |
| BLAST+       | 2.3.0                     |
| Transdecoder | 2.0.1                     |
| cd-hit       | CD-HIT version 4.6        |
| HMMeR        | 3.1b2                     |

## Accession Numbers

BioProject: PRJNA344880  
TSA: GFAT00000000  

## References


CDHIT:  
"Clustering of highly homologous sequences to reduce thesize of large protein database", Weizhong Li, Lukasz Jaroszewski & Adam Godzik. Bioinformatics, (2001) 17:282-283  
   "Cd-hit: a fast program for clustering and comparing large sets of protein or nucleotide sequences", Weizhong Li & Adam Godzik. Bioinformatics, (2006) 22:1658-1659  

HMMER:  
Accelerated profile HMM searches. S. R. Eddy. PLoS Comp. Biol., 7:e1002195, 2011.  

Trinity and Transdecoder:  
Grabherr MG, Haas BJ, Yassour M, Levin JZ, Thompson DA, Amit I, Adiconis X, Fan L, Raychowdhury R, Zeng Q, Chen Z, Mauceli E, Hacohen N, Gnirke A, Rhind N, di Palma F, Birren BW, Nusbaum C, Lindblad-Toh K, Friedman N, Regev A. Full-length transcriptome assembly from RNA-seq data without a reference genome. Nat Biotechnol. 2011 May 15;29(7):644-52. doi: 10.1038/nbt.1883. PubMed PMID: 21572440.  

BLAST+:  
Camacho C., Coulouris G., Avagyan V., Ma N., Papadopoulos J., Bealer K., & Madden T.L. (2008) "BLAST+: architecture and applications." BMC Bioinformatics 10:421.  

Seqclean:  
https://sourceforge.net/projects/seqclean/  



```{r session, echo=FALSE}
writeLines(capture.output(sessionInfo()), "sessionInfo/transcriptomeSessionInfo.txt")
```

#[R-session information](`r paste0("http://bioinfo/", gsub("/home/", "~", getwd()), "/sessionInfo/StorySessionInfo.txt")`)

</div>

<p>
<a href="http://stowers.org"><img src="images/Stowers-logo_001.jpg"></a> 
<a href="http://planaria.stowers.org/"><img src="http://smedgd.stowers.org/wp-content/uploads/2015/11/sllogo_whitebg_small.png"></a> 
<a href="http://www.hhmi.org/"><img src="images/HHMI-logo-color_small.jpg"></a>
<a href="http://www.mbl.edu/"><img src="images/marine-biological-laboratory_small.jpg"></a>
</p>



<p><a href="http://www.stowers.org/privacy-policy">Privacy Policy</a></p>
<p><a href="http://www.stowers.org/terms">Terms of Use</a></p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
