<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Eric Ross" />


<title>Transcriptome</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/sandstone.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<link rel="stylesheet"
      href="site_libs/highlight/textmate.css"
      type="text/css" />
<script src="site_libs/highlight/highlight.js"></script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>

<link rel="stylesheet" href="styles.css" type="text/css" />

</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 61px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 66px;
  margin-top: -66px;
}

.section h2 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h3 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h4 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h5 {
  padding-top: 66px;
  margin-top: -66px;
}
.section h6 {
  padding-top: 66px;
  margin-top: -66px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.9em;
  padding-left: 5px;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
  padding-left: 10px;
}

</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"><em>Mnemiopsis leidyi</em></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="experiment.html">Experiment</a>
</li>
<li>
  <a href="transcriptome.html">Transcriptome</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Quality Control
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="analysis8hpaAllR.html">8hpa Control First Pass</a>
    </li>
    <li>
      <a href="analysis8hpa5RR.html">8hpa Control Second Pass</a>
    </li>
    <li>
      <a href="analysis2hpaAllR.html">2hpa Control First Pass</a>
    </li>
    <li>
      <a href="analysis2hpa2RR.html">2hpa Control Second Pass</a>
    </li>
    <li>
      <a href="analysisStepwise.html">Stepwise Test</a>
    </li>
  </ul>
</li>
<li>
  <a href="StoryOfInitialAnalysis.html">Differential Expression</a>
</li>
<li>
  <a href="downloads.html">Downloads</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Transcriptome</h1>
<h4 class="author"><em>Eric Ross</em></h4>

</div>


<div id="overview" class="section level1">
<h1>Overview</h1>
<p>Assembly and annotation of <em>Mnemiopsis leidyi</em> transcriptome.</p>
</div>
<div id="data-sources" class="section level1">
<h1>Data Sources</h1>
<table>
<thead>
<tr class="header">
<th>Time point</th>
<th>Year Collected</th>
<th>Replicate Name</th>
<th>left(1)/right(2)</th>
<th>Filename</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1hpf</td>
<td>2012</td>
<td>N7</td>
<td>1</td>
<td>1hpf_2012_N7_1.fastq.gz</td>
</tr>
<tr class="even">
<td>1hpf</td>
<td>2012</td>
<td>N7</td>
<td>2</td>
<td>1hpf_2012_N7_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>1hpf</td>
<td>2013</td>
<td>21</td>
<td>1</td>
<td>1hpf_2013_21_1.fastq.gz</td>
</tr>
<tr class="even">
<td>1hpf</td>
<td>2013</td>
<td>21</td>
<td>2</td>
<td>1hpf_2013_21_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>1hpf</td>
<td>2013</td>
<td>41</td>
<td>1</td>
<td>1hpf_2013_41_1.fastq.gz</td>
</tr>
<tr class="even">
<td>1hpf</td>
<td>2013</td>
<td>41</td>
<td>2</td>
<td>1hpf_2013_41_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>2hpf</td>
<td>2012</td>
<td>N6</td>
<td>1</td>
<td>2hpf_2012_N6_1.fastq.gz</td>
</tr>
<tr class="even">
<td>2hpf</td>
<td>2012</td>
<td>N6</td>
<td>2</td>
<td>2hpf_2012_N6_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>2hpf</td>
<td>2013</td>
<td>22</td>
<td>1</td>
<td>2hpf_2013_22_1.fastq.gz</td>
</tr>
<tr class="even">
<td>2hpf</td>
<td>2013</td>
<td>22</td>
<td>2</td>
<td>2hpf_2013_22_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>2hpf</td>
<td>2013</td>
<td>32</td>
<td>1</td>
<td>2hpf_2013_32_1.fastq.gz</td>
</tr>
<tr class="even">
<td>2hpf</td>
<td>2013</td>
<td>32</td>
<td>2</td>
<td>2hpf_2013_32_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>2hpf</td>
<td>2013</td>
<td>42</td>
<td>1</td>
<td>2hpf_2013_42_1.fastq.gz</td>
</tr>
<tr class="even">
<td>2hpf</td>
<td>2013</td>
<td>42</td>
<td>2</td>
<td>2hpf_2013_42_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>3hpf</td>
<td>2012</td>
<td>N5</td>
<td>1</td>
<td>3hpf_2012_N5_1.fastq.gz</td>
</tr>
<tr class="even">
<td>3hpf</td>
<td>2012</td>
<td>N5</td>
<td>2</td>
<td>3hpf_2012_N5_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>3hpf</td>
<td>2013</td>
<td>23</td>
<td>1</td>
<td>3hpf_2013_23_1.fastq.gz</td>
</tr>
<tr class="even">
<td>3hpf</td>
<td>2013</td>
<td>23</td>
<td>2</td>
<td>3hpf_2013_23_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>3hpf</td>
<td>2013</td>
<td>33</td>
<td>1</td>
<td>3hpf_2013_33_1.fastq.gz</td>
</tr>
<tr class="even">
<td>3hpf</td>
<td>2013</td>
<td>33</td>
<td>2</td>
<td>3hpf_2013_33_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>4hpf</td>
<td>2012</td>
<td>N4</td>
<td>1</td>
<td>4hpf_2012_N4_1.fastq.gz</td>
</tr>
<tr class="even">
<td>4hpf</td>
<td>2012</td>
<td>N4</td>
<td>2</td>
<td>4hpf_2012_N4_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>4hpf</td>
<td>2013</td>
<td>24</td>
<td>1</td>
<td>4hpf_2013_24_1.fastq.gz</td>
</tr>
<tr class="even">
<td>4hpf</td>
<td>2013</td>
<td>24</td>
<td>2</td>
<td>4hpf_2013_24_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>4hpf</td>
<td>2013</td>
<td>34</td>
<td>1</td>
<td>4hpf_2013_34_1.fastq.gz</td>
</tr>
<tr class="even">
<td>4hpf</td>
<td>2013</td>
<td>34</td>
<td>2</td>
<td>4hpf_2013_34_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>5hpf</td>
<td>2012</td>
<td>14</td>
<td>1</td>
<td>5hpf_2012_14_1.fastq.gz</td>
</tr>
<tr class="even">
<td>5hpf</td>
<td>2012</td>
<td>14</td>
<td>2</td>
<td>5hpf_2012_14_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>5hpf</td>
<td>2012</td>
<td>N3</td>
<td>1</td>
<td>5hpf_2012_N3_1.fastq.gz</td>
</tr>
<tr class="even">
<td>5hpf</td>
<td>2012</td>
<td>N3</td>
<td>2</td>
<td>5hpf_2012_N3_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>5hpf</td>
<td>2013</td>
<td>25</td>
<td>1</td>
<td>5hpf_2013_25_1.fastq.gz</td>
</tr>
<tr class="even">
<td>5hpf</td>
<td>2013</td>
<td>25</td>
<td>2</td>
<td>5hpf_2013_25_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>6hpf</td>
<td>2012</td>
<td>23</td>
<td>1</td>
<td>6hpf_2012_23_1.fastq.gz</td>
</tr>
<tr class="even">
<td>6hpf</td>
<td>2012</td>
<td>23</td>
<td>2</td>
<td>6hpf_2012_23_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>6hpf</td>
<td>2012</td>
<td>N2</td>
<td>1</td>
<td>6hpf_2012_N2_1.fastq.gz</td>
</tr>
<tr class="even">
<td>6hpf</td>
<td>2012</td>
<td>N2</td>
<td>2</td>
<td>6hpf_2012_N2_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>6hpf</td>
<td>2013</td>
<td>26</td>
<td>1</td>
<td>6hpf_2013_26_1.fastq.gz</td>
</tr>
<tr class="even">
<td>6hpf</td>
<td>2013</td>
<td>26</td>
<td>2</td>
<td>6hpf_2013_26_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>6hpf</td>
<td>2013</td>
<td>46</td>
<td>1</td>
<td>6hpf_2013_46_1.fastq.gz</td>
</tr>
<tr class="even">
<td>6hpf</td>
<td>2013</td>
<td>46</td>
<td>2</td>
<td>6hpf_2013_46_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>7hpf</td>
<td>2012</td>
<td>N1</td>
<td>1</td>
<td>7hpf_2012_N1_1.fastq.gz</td>
</tr>
<tr class="even">
<td>7hpf</td>
<td>2012</td>
<td>N1</td>
<td>2</td>
<td>7hpf_2012_N1_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>7hpf</td>
<td>2013</td>
<td>17</td>
<td>1</td>
<td>7hpf_2013_17_1.fastq.gz</td>
</tr>
<tr class="even">
<td>7hpf</td>
<td>2013</td>
<td>17</td>
<td>2</td>
<td>7hpf_2013_17_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>7hpf</td>
<td>2013</td>
<td>27</td>
<td>1</td>
<td>7hpf_2013_27_1.fastq.gz</td>
</tr>
<tr class="even">
<td>7hpf</td>
<td>2013</td>
<td>27</td>
<td>2</td>
<td>7hpf_2013_27_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>7hpf</td>
<td>2013</td>
<td>47</td>
<td>1</td>
<td>7hpf_2013_47_1.fastq.gz</td>
</tr>
<tr class="even">
<td>7hpf</td>
<td>2013</td>
<td>47</td>
<td>2</td>
<td>7hpf_2013_47_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>8hpf</td>
<td>2012</td>
<td>18</td>
<td>1</td>
<td>8hpf_2012_18_1.fastq.gz</td>
</tr>
<tr class="even">
<td>8hpf</td>
<td>2012</td>
<td>18</td>
<td>2</td>
<td>8hpf_2012_18_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>8hpf</td>
<td>2012</td>
<td>N8B</td>
<td>1</td>
<td>8hpf_2012_N8B_1.fastq.gz</td>
</tr>
<tr class="even">
<td>8hpf</td>
<td>2012</td>
<td>N8B</td>
<td>2</td>
<td>8hpf_2012_N8B_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>8hpf</td>
<td>2013</td>
<td>18</td>
<td>1</td>
<td>8hpf_2013_18_1.fastq.gz</td>
</tr>
<tr class="even">
<td>8hpf</td>
<td>2013</td>
<td>18</td>
<td>2</td>
<td>8hpf_2013_18_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>8hpf</td>
<td>2013</td>
<td>28</td>
<td>1</td>
<td>8hpf_2013_28_1.fastq.gz</td>
</tr>
<tr class="even">
<td>8hpf</td>
<td>2013</td>
<td>28</td>
<td>2</td>
<td>8hpf_2013_28_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>8hpf</td>
<td>2013</td>
<td>48</td>
<td>1</td>
<td>8hpf_2013_48_1.fastq.gz</td>
</tr>
<tr class="even">
<td>8hpf</td>
<td>2013</td>
<td>48</td>
<td>2</td>
<td>8hpf_2013_48_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>9hpf</td>
<td>2012</td>
<td>37</td>
<td>1</td>
<td>9hpf_2012_37_1.fastq.gz</td>
</tr>
<tr class="even">
<td>9hpf</td>
<td>2012</td>
<td>37</td>
<td>2</td>
<td>9hpf_2012_37_2.fastq.gz</td>
</tr>
<tr class="odd">
<td>9hpf</td>
<td>2013</td>
<td>49</td>
<td>1</td>
<td>9hpf_2013_49_1.fastq.gz</td>
</tr>
<tr class="even">
<td>9hpf</td>
<td>2013</td>
<td>49</td>
<td>2</td>
<td>9hpf_2013_49_2.fastq.gz</td>
</tr>
</tbody>
</table>
</div>
<div id="assembly" class="section level1">
<h1>Assembly</h1>
<p>Left and right reads were separately concatenated and then assembled with Trinity version trinityrnaseq_r20140717 <span class="citation">(Grabherr et al. 2011)</span>.</p>
<pre><code>Trinity \  
--normalize_reads \  
--max_memory 500G \  
--CPU 16 \  
--SS_lib_type FR \  
--left \  
lefts.fastq \  
--right \  
rights.fastq  
</code></pre>
<div id="initial-assembly-statistics" class="section level2">
<h2>Initial Assembly Statistics</h2>
<table>
<tbody>
<tr class="odd">
<td>Number of Sequences:</td>
<td align="right">233,327</td>
</tr>
<tr class="even">
<td>Total Length:</td>
<td align="right">160,677,750</td>
</tr>
<tr class="odd">
<td>Average Length:</td>
<td align="right">688</td>
</tr>
<tr class="even">
<td>Longest Sequence:</td>
<td align="right">29,348</td>
</tr>
<tr class="odd">
<td>Shortest Sequence:</td>
<td align="right">201</td>
</tr>
<tr class="even">
<td>%GC:</td>
<td align="right">40%</td>
</tr>
<tr class="odd">
<td>N50:</td>
<td align="right">1,152</td>
</tr>
<tr class="even">
<td>N90:</td>
<td align="right">266</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="sequence-cleaning" class="section level1">
<h1>Sequence Cleaning</h1>
<p>Adapter and contaminate was removed from assembly using SeqClean <span class="citation">(Cleaner 2017)</span>. Additionally, sequences shorter than 300 bp were discarded.</p>
<pre><code>seqclean \  
Mnemiopsis_leidyi_trinity_20140802_unfiltered.nt \  
-v UniVec \  
-l 300 \  
-c 16 \  
-s Hs_GRCh38.nt,ecoli.nt  
</code></pre>
<pre><code>Output:
**************************************************
Sequences analyzed:    233327
-----------------------------------
                   valid:    141346  (24371 trimmed)
                 trashed:     91981
**************************************************
----= Trashing summary =------
               by 'short':    90500
        by 'Hs_GRCh38.nt':       11
              by 'shortq':      140
            by 'ecoli.nt':       17
              by 'UniVec':     1313
------------------------------
</code></pre>
<div id="cleaned-assembly-statistics" class="section level2">
<h2>Cleaned Assembly Statistics</h2>
<table>
<tbody>
<tr class="odd">
<td>Number of Sequences:</td>
<td align="right">141,346</td>
</tr>
<tr class="even">
<td>Total Length:</td>
<td align="right">137,906,212</td>
</tr>
<tr class="odd">
<td>Average Length:</td>
<td align="right">975</td>
</tr>
<tr class="even">
<td>Longest Sequence:</td>
<td align="right">29,348</td>
</tr>
<tr class="odd">
<td>Shortest Sequence:</td>
<td align="right">300</td>
</tr>
<tr class="even">
<td>%GC:</td>
<td align="right">41%</td>
</tr>
<tr class="odd">
<td>N50:</td>
<td align="right">1,486</td>
</tr>
<tr class="even">
<td>N90:</td>
<td align="right">405</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="genome-filtering" class="section level1">
<h1>Genome Filtering</h1>
<p>In order to produce a more accurate transcriptome set, we filtered out sequences not present in the available <em>M. leidyi</em> genome. We used the very permissive requirement that any transcript aligning with at least 90% identity over any length of genomic sequence ought to be retained.</p>
<div id="genome-filtered-transcript-statistics" class="section level2">
<h2>Genome Filtered Transcript Statistics</h2>
<table>
<tbody>
<tr class="odd">
<td>Number of Sequences:</td>
<td align="right">44,438</td>
</tr>
<tr class="even">
<td>Total Length:</td>
<td align="right">62,917,788</td>
</tr>
<tr class="odd">
<td>Average Length:</td>
<td align="right">1,415</td>
</tr>
<tr class="even">
<td>Longest Sequence:</td>
<td align="right">29,348</td>
</tr>
<tr class="odd">
<td>Shortest Sequence:</td>
<td align="right">300</td>
</tr>
<tr class="even">
<td>%GC:</td>
<td align="right">41%</td>
</tr>
<tr class="odd">
<td>%N:</td>
<td align="right">0%</td>
</tr>
<tr class="even">
<td>N50:</td>
<td align="right">2,168</td>
</tr>
<tr class="odd">
<td>N90:</td>
<td align="right">601</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="additional-sequence-cleaning" class="section level1">
<h1>Additional Sequence Cleaning</h1>
<p>NCBI detected additional contaminate in our sequences, primarily vector/adapter. We sequentially ran SeqClean to remove concatenated adapter.</p>
<div id="cleaned-sequence-statistics" class="section level2">
<h2>Cleaned Sequence Statistics</h2>
<table>
<tbody>
<tr class="odd">
<td>Number of Sequences:</td>
<td align="right">44,388</td>
</tr>
<tr class="even">
<td>Total Length:</td>
<td align="right">62,859,653</td>
</tr>
<tr class="odd">
<td>Average Length:</td>
<td align="right">1,416</td>
</tr>
<tr class="even">
<td>Longest Sequence:</td>
<td align="right">29,348</td>
</tr>
<tr class="odd">
<td>Shortest Sequence:</td>
<td align="right">300</td>
</tr>
<tr class="even">
<td>%GC:</td>
<td align="right">41%</td>
</tr>
<tr class="odd">
<td>%N:</td>
<td align="right">0%</td>
</tr>
<tr class="even">
<td>N50:</td>
<td align="right">2,167</td>
</tr>
<tr class="odd">
<td>N90:</td>
<td align="right">601</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="transcript-collapse" class="section level1">
<h1>Transcript collapse</h1>
<p>For RNAseq we wished to create a non-redundant set of transcripts by collapsing isoforms using CD-HIT <span class="citation">(Li, Jaroszewski, and Godzik 2001)</span>.</p>
<pre><code>cd-hit-est \  
-i in_genome_clean_r6.nt \  
-o cdhit.nt \  
-r 0 \  
-c .95 \  
-G 0 \  
-aS .5 \  
-M 0 \  
-T 0  
</code></pre>
<div id="non-redundant-sequence-statistics" class="section level2">
<h2>Non-redundant Sequence Statistics</h2>
<table>
<tbody>
<tr class="odd">
<td>Number of Sequences:</td>
<td align="right">31,067</td>
</tr>
<tr class="even">
<td>Total Length:</td>
<td align="right">47,344,147</td>
</tr>
<tr class="odd">
<td>Average Length:</td>
<td align="right">1,523</td>
</tr>
<tr class="even">
<td>Longest Sequence:</td>
<td align="right">29,348</td>
</tr>
<tr class="odd">
<td>Shortest Sequence:</td>
<td align="right">300</td>
</tr>
<tr class="even">
<td>%GC:</td>
<td align="right">41%</td>
</tr>
<tr class="odd">
<td>%N:</td>
<td align="right">0%</td>
</tr>
<tr class="even">
<td>N50:</td>
<td align="right">2,200</td>
</tr>
<tr class="odd">
<td>N90:</td>
<td align="right">692</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="sequence-renaming" class="section level1">
<h1>Sequence Renaming</h1>
<p>Sequences were renamed with a “Mle” prefix and zero padded numbers. Names are in the format of ‘Mle_000000’.</p>
</div>
<div id="annotation" class="section level1">
<h1>Annotation</h1>
<div id="swissprot-uniprot" class="section level3">
<h3>Swissprot / Uniprot</h3>
<p>Best BLASTX hits to Swissprot with a maximum evalue of 0.001 were used to annotate genes <span class="citation">(Camacho et al. 2009)</span>.</p>
<pre><code>blastx \
-query Mleidyi_20160525.nt \  
-db swissprot \  
-max_target_seqs 1 \  
-max_hsps 1 \  
-evalue .001 \  
-outfmt 6 \  
-out ml_v_sp.blastx \  
-num_threads 32
</code></pre>
</div>
<div id="pfam" class="section level3">
<h3>Pfam</h3>
Sequences were translated with Transdecoder and then domains were assigned to transcripts using hmmscan against the Pfam-A database <span class="citation">(Eddy 2011)</span>.<br />

<pre><code>Transdecoder.LongOrfs \  
-S \
-m 60 \
-t Mleidyi_20160525.aa.fa
</code></pre>
<p>All ORFs of 60aa or greater were retained.</p>
<pre><code>hmmscan \  
-o Mleidyi_20160525.aa.fa.pfam.out \  
--domtblout Mleidyi_20160525.aa.fa.pfam.table \  
-E .01 \  
--domE .01 \  
--cpu 16 \  
../db/Pfam-A.hmm \  
</code></pre>
<pre><code>
awk '{print $4 "\t" $2}' Mleidyi_20160525.aa.fa.pfam.table | perl -p -e 's/(Mle_\d+)\|.*(\t.+)/$1$2/' | sort -u > mleidyi2pfam.txt
</code></pre>
</div>
</div>
<div id="software" class="section level1">
<h1>Software</h1>
<table>
<thead>
<tr class="header">
<th>Program</th>
<th>Version</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Trinity</td>
<td>trinityrnaseq_r20140717</td>
</tr>
<tr class="even">
<td>Seqclean</td>
<td>Downloaded Feb 22, 2011</td>
</tr>
<tr class="odd">
<td>BLAST+</td>
<td>2.3.0</td>
</tr>
<tr class="even">
<td>Transdecoder</td>
<td>2.0.1</td>
</tr>
<tr class="odd">
<td>cd-hit</td>
<td>CD-HIT version 4.6</td>
</tr>
<tr class="even">
<td>HMMeR</td>
<td>3.1b2</td>
</tr>
</tbody>
</table>
</div>
<div id="accession-numbers" class="section level1">
<h1>Accession Numbers</h1>
<p>BioProject: PRJNA344880<br />
TSA: GFAT00000000</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1>References</h1>
<div id="refs" class="references">
<div id="ref-Camacho2009">
<p>Camacho, Christiam, George Coulouris, Vahram Avagyan, Ning Ma, Jason Papadopoulos, Kevin Bealer, and Thomas L Madden. 2009. “BLAST+: Architecture and Applications.” <em>BMC Bioinformatics</em> 10 (1). Springer Nature: 421. doi:<a href="https://doi.org/10.1186/1471-2105-10-421">10.1186/1471-2105-10-421</a>.</p>
</div>
<div id="ref-1_cleaner_2017">
<p>Cleaner. 2017. “Sequence Cleaner.” <em>SourceForge</em>. <a href="https://sourceforge.net/projects/seqclean/" class="uri">https://sourceforge.net/projects/seqclean/</a>.</p>
</div>
<div id="ref-Eddy2011">
<p>Eddy, Sean R. 2011. “Accelerated Profile HMM Searches.” Edited by William R. Pearson. <em>PLoS Computational Biology</em> 7 (10). Public Library of Science (PLoS): e1002195. doi:<a href="https://doi.org/10.1371/journal.pcbi.1002195">10.1371/journal.pcbi.1002195</a>.</p>
</div>
<div id="ref-Grabherr2011">
<p>Grabherr, Manfred G, Brian J Haas, Moran Yassour, Joshua Z Levin, Dawn A Thompson, Ido Amit, Xian Adiconis, et al. 2011. “Full-Length Transcriptome Assembly from RNA-Seq Data Without a Reference Genome.” <em>Nature Biotechnology</em> 29 (7). Springer Nature: 644–52. doi:<a href="https://doi.org/10.1038/nbt.1883">10.1038/nbt.1883</a>.</p>
</div>
<div id="ref-Li2001">
<p>Li, W., L. Jaroszewski, and A. Godzik. 2001. “Clustering of Highly Homologous Sequences to Reduce the Size of Large Protein Databases.” <em>Bioinformatics</em> 17 (3). Oxford University Press (OUP): 282–83. doi:<a href="https://doi.org/10.1093/bioinformatics/17.3.282">10.1093/bioinformatics/17.3.282</a>.</p>
</div>
</div>
</div>


<p>
<a href="http://stowers.org"><img src="images/Stowers-logo_001.jpg" style="width:200px;height:75px;"></a> 
<a href="http://planaria.stowers.org/"><img src="http://smedgd.stowers.org/wp-content/uploads/2015/11/sllogo_whitebg_small.png"></a> 
<a href="http://www.hhmi.org/"><img src="images/HHMI-logo-color_small.jpg"></a>
<a href="http://www.mbl.edu/"><img src="images/marine-biological-laboratory_small.jpg"></a>
</p>



<p><a href="http://www.stowers.org/privacy-policy">Privacy Policy</a></p>
<p><a href="http://www.stowers.org/terms">Terms of Use</a></p>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
$(document).ready(function () {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
